File: backend/app/schemas/enrolment.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class EnrolmentResponse(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    student_id: uuid.UUID
    status: str
    enrolled_at: datetime
    opted_out_at: datetime | None
    session_title: str | None = None
    tutor_name: str | None = None

    model_config = {"from_attributes": True}

File: mobile/src/types/enrolment.ts
import { EnrolmentStatus } from './common';

export interface EnrolmentResponse {
  id: string;
  session_id: string;
  student_id: string;
  status: EnrolmentStatus;
  enrolled_at: string;
  opted_out_at: string | null;
  session_title: string | null;
  tutor_name: string | null;
}
